pkg/validators/am0011_sku_validation.go

```diff
@@ -0,0 +1,64 @@
+package validators
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/mt-sre/addon-metadata-operator/pkg/types"
+	"github.com/mt-sre/addon-metadata-operator/pkg/utils"
+)
+
+var AM0011 = types.Validator{
+	Code:        "AM0011",
+	Name:        "sku_validation",
+	Description: "Validates whether a SKU Rule exists in OCM for quota provided in addon metadata",
+	Runner:      ValidateOCMSKUExists,
+}
+
+func init() {
+	Registry.Add(AM0011)
+}
+
+func ValidateOCMSKUExists(mb types.MetaBundle) types.ValidatorResult {
+	client, err := utils.NewDefaultOCMClient()
+	if err != nil {
+		return Error(err)
+	}
+
+	validateFunc := GenerateOCMSKUValidator(client)
```

Comment:
Why use a func generator? Seems the only effect is to have code indented 8 spaces because of the `return func() { <definition> }`

Reply:
Edit: saw it's to mock the client in testing. Clever ! :heart: